src/app/loginComponents/requests/requests.component.ts

```diff
@@ -118,4 +122,28 @@ export class RequestsComponent extends Base implements OnInit {
   rerequestReview(membership: OrganizationUser) {
     this.requestsService.requestRereview(membership.organization.id);
   }
+
+  removeOrganizationDialog(organizationName: string, organizationStatus: string, organizationID: number) {
+    const confirmationDialogData: ConfirmationDialogData = {
+      title: 'Delete Organization',
+      message: `Are you sure you wish to delete this ${organizationStatus.toLowerCase()} organization?
+                All information associated with <b>${organizationName}</b> will be deleted.`,
+      cancelButtonText: 'Cancel',
+      confirmationButtonText: 'Delete'
+    };
+    this.confirmationDialogService
+      .openDialog(confirmationDialogData, bootstrap4mediumModalSize)
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe(result => {
+        if (result) {
+          this.deleteOrganization(organizationID);
+        }
+      });
+  }
+
+  deleteOrganization(organizationID: number) {
+    this.userQuery.isAdminOrCurator$.pipe(takeUntil(this.ngUnsubscribe)).subscribe(isAdminOrCurator => {
```

Comment:
- Don't subscribe here, causes issue if state is refreshed.
- adminOrCurator check before the confirmation dialog pops up.  Another possibility, disable the button if they're not an admin/curator

Reply:
Removed the method entirely and created a boolean field to reflect the admin/curator status. The new field is assigned a value at initialization when the code decides to update the curator orgs or not.

Checking for the admin/curator status is to prevent updating the curators' orgs else an error would be thrown if the user is not an admin/curator. Re-requesting organizations seems to be running into this issue as well; will be creating a ticket for that.